diff --git a/youtube_dl/extractor/extractors.py b/youtube_dl/extractor/extractors.py
index ab8d6a5a5..556b632e4 100644
--- a/youtube_dl/extractor/extractors.py
+++ b/youtube_dl/extractor/extractors.py
@@ -773,7 +773,7 @@ from .nick import (
     NickNightIE,
     NickRuIE,
 )
-from .niconico import NiconicoIE, NiconicoPlaylistIE
+from .niconico import NiconicoIE, NiconicoPlaylistIE, NiconicoShortIE
 from .ninecninemedia import NineCNineMediaIE
 from .ninegag import NineGagIE
 from .ninenow import NineNowIE
diff --git a/youtube_dl/extractor/niconico.py b/youtube_dl/extractor/niconico.py
index a85fc3d5c..d57d6eae5 100644
--- a/youtube_dl/extractor/niconico.py
+++ b/youtube_dl/extractor/niconico.py
@@ -513,3 +513,8 @@ class NiconicoPlaylistIE(InfoExtractor):
             entries, list_id, mylist.get('name'), mylist.get('description'))
         result.update(self._parse_owner(mylist))
         return result
+
+
+class NiconicoShortIE(NiconicoIE):
+    _VALID_URL = r'(?P<id>^(sm|nm|so)[0-9]+)'
+pass
\ No newline at end of file
