diff --git a/musicbot/downloader.py b/musicbot/downloader.py
index 51a36e0..76cfcf1 100644
--- a/musicbot/downloader.py
+++ b/musicbot/downloader.py
@@ -2,8 +2,9 @@ import os
 import asyncio
 import logging
 import functools
-import youtube_dl
+#import youtube_dl
 import subprocess, re, time
+import yt_dlp as youtube_dl
 
 import concurrent.futures
 from concurrent.futures import ThreadPoolExecutor
@@ -47,18 +48,22 @@ youtube_dl.utils.bug_reports_message = lambda: ''
 class Downloader:
     def __init__(self, download_folder=None):
         self.thread_pool = ThreadPoolExecutor(max_workers=4)
-        self.unsafe_ytdl = youtube_dl.YoutubeDL(ytdl_format_options)
-        self.safe_ytdl = youtube_dl.YoutubeDL(ytdl_format_options)
-        self.safe_ytdl.params['ignoreerrors'] = True
+        #self.unsafe_ytdl = youtube_dl.YoutubeDL(ytdl_format_options)
+        #self.safe_ytdl = youtube_dl.YoutubeDL(ytdl_format_options)
+        #self.safe_ytdl.params['ignoreerrors'] = True
         self.download_folder = download_folder
 
         if download_folder:
-            otmpl = self.unsafe_ytdl.params['outtmpl']
-            self.unsafe_ytdl.params['outtmpl'] = os.path.join(download_folder, otmpl)
-            # print("setting template to " + os.path.join(download_folder, otmpl))
-
-            otmpl = self.safe_ytdl.params['outtmpl']
-            self.safe_ytdl.params['outtmpl'] = os.path.join(download_folder, otmpl)
+            #otmpl = self.unsafe_ytdl.params['outtmpl']
+            otmpl = ytdl_format_options["outtmpl"]
+            #self.unsafe_ytdl.params['outtmpl'] = os.path.join(download_folder, otmpl)
+            ytdl_format_options["outtmpl"] = os.path.join(download_folder, otmpl)
+            print("setting template to " + os.path.join(download_folder, otmpl))
+
+            #otmpl = self.safe_ytdl.params['outtmpl']
+            #self.safe_ytdl.params['outtmpl'] = os.path.join(download_folder, otmpl)
+            self.unsafe_ytdl = youtube_dl.YoutubeDL(ytdl_format_options)
+            self.safe_ytdl = youtube_dl.YoutubeDL( {**ytdl_format_options, "ignoreerrors": True} )
 
 
     @property
diff --git a/musicbot/playlist.py b/musicbot/playlist.py
index 938115e..5ab534e 100644
--- a/musicbot/playlist.py
+++ b/musicbot/playlist.py
@@ -7,7 +7,8 @@ from itertools import islice
 from collections import deque
 
 from urllib.error import URLError
-from youtube_dl.utils import ExtractorError, DownloadError, UnsupportedError
+#from youtube_dl.utils import ExtractorError, DownloadError, UnsupportedError
+from yt_dlp.utils import ExtractorError, DownloadError, UnsupportedError
 
 from .utils import get_header
 from .constructs import Serializable
